libs/sdk-ui-kit/src/@ui/UiMenu/itemUtils.ts

```
@@ -0,0 +1,138 @@
+    for (const item of items) {
+        if (predicate(item)) {
+            return item;
+        }
+
+        if (item.type === "interactive" && item.subMenu) {
+            const foundItemInSubMenu = findItem(item.subMenu, predicate);
+
+            if (foundItemInSubMenu) {
+                return foundItemInSubMenu;
+            }
+        }
+    }
+    return undefined;
+};
+
+export const getItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
+): IUiMenuItem<InteractiveItemData, StaticItemData> | undefined => {
+    return findItem(items, (item) => item.id === id);
+};
+
+export const findInteractiveItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    predicate: (item: IUiMenuInteractiveItem<InteractiveItemData, StaticItemData>) => boolean,
+): IUiMenuInteractiveItem<InteractiveItemData, StaticItemData> | undefined => {
+    const foundItem = findItem(items, (item) => item.type === "interactive" && predicate(item));
+    return foundItem?.type === "interactive" ? foundItem : undefined;
+};
+
+export const getInteractiveItem = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
+): IUiMenuInteractiveItem<InteractiveItemData, StaticItemData> | undefined => {
+    return findInteractiveItem(items, (item) => item.id === id);
+};
+
+export const getItemsInSubMenu = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
+): IUiMenuItem<InteractiveItemData, StaticItemData>[] | undefined => {
+    return id == null ? items : findInteractiveItem(items, (item) => item.id === id)?.subMenu;
+};
+
+export const getItemParent = <InteractiveItemData, StaticItemData>(
+    items: IUiMenuItem<InteractiveItemData, StaticItemData>[],
+    id?: string,
```

fail fast if id === undefined